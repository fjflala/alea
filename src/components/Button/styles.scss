.ui-button {
  display: block;
  height: 48px;
  width: 100%;
  max-width: 230px;
  font-size: 20px;
  font-weight: bold;
  text-decoration: none;
  text-align: center;
  color: #fff;
  border-radius: 24px;
  cursor: pointer;

  &--primary {
    background-color: rgb(245,85,46);
  }

  &--secondary {
    background-color: white;
    color: rgb(245,85,46);
    border: 1px solid #eee;
  }
}