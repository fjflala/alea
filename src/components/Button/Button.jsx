/**
 * Module dependencies
 */
import React from 'react';
import PropTypes from 'prop-types';

/**
 * Import styles
 */
import './styles.scss';

/**
 * Represents a Button
 * @param {object} props - Props of the element
 * @param {function} props.onClick - Callback when element is clicked
 * @param {string} props.type - Type of button
 * @param {node} props.children - Children element
 * @param {string} props.className - Custom className of the button
 * @param {string} props.modifier - Modifiers for the element
 */
const Button = ({
  onClick,
  type,
  children,
  className,
  modifier,
}) => {
  const cn = `ui-button ${modifier ? `ui-button--${modifier}` : ''} ${className && className}`;
  return (
    <button
      type={type}
      className={cn}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

/**
 * PropTypes definitions
 */
Button.propTypes = {
  onClick: PropTypes.func,
  type: PropTypes.oneOf(['submit', 'button', 'reset']),
  modifier: PropTypes.oneOf(['primary', 'secondary']),
  className: PropTypes.string,
};

Button.defaultProps = {
  onClick: () => {},
  type: 'button',
  modifier: 'primary',
  className: '',
};

/**
 * Expose element
 */
export default Button;
