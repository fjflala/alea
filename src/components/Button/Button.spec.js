import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('<Button />', () => {
  it('should correctly render', () => {
    const props = {
      modifier: 'primary',
      onClick: jest.fn(),
      className: 'custom-class',
    };

    const tree = renderer
      .create(<Button {...props} />)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});