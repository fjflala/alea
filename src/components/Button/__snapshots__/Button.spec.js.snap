// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should correctly render 1`] = `
<button
  className="ui-button ui-button--primary custom-class"
  onClick={[MockFunction]}
  type="button"
/>
`;
